utility/session_test.go

```diff
@@ -0,0 +1,535 @@
+
+			// Assert different fishermen between heights for the same app
+			assertActorsDifference(t, app1PrevFishermen, session1.Fishermen, probabilityOfOverlap)
+			assertActorsDifference(t, app2PrevFishermen, session2.Fishermen, probabilityOfOverlap)
+			assertActorsDifference(t, app3PrevFishermen, session3.Fishermen, probabilityOfOverlap)
+
+			// Store the new servicers and fishermen for the next height
+			app1PrevServicers = session1.Servicers
+			app2PrevServicers = session2.Servicers
+			app3PrevServicers = session3.Servicers
+			app1PrevFishermen = session1.Fishermen
+			app2PrevFishermen = session2.Fishermen
+			app3PrevFishermen = session3.Fishermen
+
+			// Reset the number of blocks until we expect a different set of servicers and fishermen
+			numBlocksUntilChange = numBlocksPerSession - 1
+		} else {
+			// Assert the same servicers between heights for the same app
+			require.ElementsMatch(t, app1PrevServicers, session1.Servicers)
+			require.ElementsMatch(t, app2PrevServicers, session2.Servicers)
+			require.ElementsMatch(t, app3PrevServicers, session3.Servicers)
+
+			// Assert the same fishermen between heights for the same app
+			require.ElementsMatch(t, app1PrevFishermen, session1.Fishermen)
+			require.ElementsMatch(t, app2PrevFishermen, session2.Fishermen)
+			require.ElementsMatch(t, app3PrevFishermen, session3.Fishermen)
+
+			numBlocksUntilChange--
+		}
+
+		// Advance block height
+		writeCtx, err := persistenceMod.NewRWContext(height)
+		require.NoError(t, err)
+		err = writeCtx.Commit([]byte(fmt.Sprintf("proposer_height_%d", height)), []byte(fmt.Sprintf("quorum_cert_height_%d", height)))
+		require.NoError(t, err)
+		writeCtx.Release()
+	}
+}
+
+func TestSession_GetSession_ApplicationUnbonds(t *testing.T) {
+	// TODO: What if an Application unbonds (unstaking period elapses) mid session?
+}
+
+func TestSession_GetSession_ServicersAndFishermenCounts_GeoZoneAvailability(t *testing.T) {
+	// TECHDEBT(#697): Once GeoZones are implemented, the tests need to be added as well
+	// Cases: Invalid, unused, non-existent, empty, insufficiently complete, etc...
+}
+
+func TestSession_GetSession_ActorReplacement(t *testing.T) {
+	// TODO: Since sessions last multiple blocks, we need to design what happens when an actor is (un)jailed, (un)stakes, (un)bonds, (un)pauses
+	// mid session. There are open design questions that need to be made.
+}
+
+func TestSession_GetSession_SessionHeightAndNumber_ModifiedBlocksPerSession(t *testing.T) {
+	// RESEARCH: Need to design what happens (actor replacement, session numbers, etc...) when the number
+	// of blocks per session changes mid session. For example, all existing sessions could go to completion
+	// until the new parameter takes effect. There are open design questions that need to be made.
+}
+
+func assertActorsDifference(t *testing.T, actors1, actors2 []*coreTypes.Actor, maxSimilarityThreshold float64) {
```

Comment:
See: https://pkg.go.dev/testing#T.Helper

```suggestion
func assertActorsDifference(t *testing.T, actors1, actors2 []*coreTypes.Actor, maxSimilarityThreshold float64) {
  t.Helper()
```